snoop/data/magic.py

```diff
@@ -82,3 +82,23 @@ def download_magic_definitions():
                 for chunk in chunks(resp):
                     f.write(chunk)
         print("ok")
+
+
+def looks_like_email(path):
+    HEADER_SET = {
+        "Relay-Version" , "Return-Path" , "From" , "To" ,
+        "Received" , "Message-Id" , "Date", "In-Reply-To", "Subject"
+    }
+    HEADER_MIN_HIT_COUNT = 2
+    HEADER_READ_SIZE = 1024 * 64
+
+    with path.open('rb') as f:
+        content = f.read(HEADER_READ_SIZE).decode('latin-1')
```

Comment:
Yeaaahhh... there's this edge case where you [might](https://docs.python.org/3/library/io.html#io.RawIOBase.read) read less than `HEADER_READ_SIZE`, even if there is more data.